lib/atom-music.js

```diff
@@ -0,0 +1,75 @@
+const AtomMusicView = require("./atom-music-view");
+const {CompositeDisposable} = require("atom");
+const configSchema = require("../config.json");
+
+const AtomMusic = {
+	atomMusicView: null,
+	modalPanel: null,
+	subscriptions: new CompositeDisposable,
+	config: configSchema,
+
+	activate(state) {
+		this.atomMusicView = new AtomMusicView(state.atomMusicViewState);
+		this.subscriptions.add(atom.workspace.addOpener(uri => {
+			if (uri === this.atomMusicView.getURI()) { return this.atomMusicView; }
+		}));
+
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:toggle": () => this.atomMusicView.toggle()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:play-pause": () => this.atomMusicView.togglePlayback()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:toggle-shuffle": () => this.atomMusicView.toggleShuffle()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:show-playlist": () => this.atomMusicView.showPlayList()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:forward-15s": () => this.atomMusicView.forward15()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:backward-15s": () => this.atomMusicView.back15()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:next-track": () => this.atomMusicView.nextTrack()}));
+		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:previous-track": () => this.atomMusicView.prevTrack()}));
+
```

Comment:
@sdinesh86 Not sure how to do, because it's default to do it this way. But your right, it doesn't look very great.
@UziTech How do you think about this part?

Reply:
```javascript
initializeSubscriptions ( ) {
this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:toggle": () => this.atomMusicView.toggle()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:play-pause": () => this.atomMusicView.togglePlayback()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:toggle-shuffle": () => this.atomMusicView.toggleShuffle()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:show-playlist": () => this.atomMusicView.showPlayList()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:forward-15s": () => this.atomMusicView.forward15()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:backward-15s": () => this.atomMusicView.back15()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:next-track": () => this.atomMusicView.nextTrack()}));
		this.subscriptions.add(atom.commands.add("atom-workspace", {"atom-music:previous-track": () => this.atomMusicView.prevTrack()}));

}

activate(state) {
		this.atomMusicView = new AtomMusicView(state.atomMusicViewState);
		this.subscriptions.add(atom.workspace.addOpener(uri => {
			if (uri === this.atomMusicView.getURI()) { return this.atomMusicView; }
		}));
               this.initializeSubscriptions();
}
```

Reply:
just would make it a bit cleaner and managable

Reply:
@sdinesh86 Hm, a little bit cleaner but not so much. 
@sdinesh86 @UziTech Would be great, it it's possible to add multiple commands and subscriptions at the same time. But because Atom is not very well documented, i don't know how.

Reply:
I have not been working with atom for quite some time now, so I am not sure about the new api.
But if i remember correctly back then there was a way to add multiple commands at the same time I dont remember it now but if get it I will let you know

Reply:
I updated it to only use a single call to `atom.commands.add`

Reply:
@UziTech Great! :-)